napari/components/viewer_model.py

```diff
@@ -133,6 +133,7 @@ class ViewerModel(KeymapProvider, MousemapProvider, EventedModel):
     title: str = 'napari'
 
     # 2-tuple indicating height and width
+    _prev_point: Tuple[int, ...] = ()
```

Comment:
Rather than storing more instance state, I think you might be able to get away with temporarily storing `Dims.point` in `ViewerModel._on_layers_change` in a similar way that `range` is temporarily stored.

```python
    def _on_layers_change(self):
        if len(self.layers) == 0:
            self.dims.ndim = 2
            self.dims.reset()
        else:
            point = self.dims.point
            ranges = self.layers._ranges
            ndim = len(ranges)
            self.dims.ndim = ndim
            self.dims.set_range(range(ndim), ranges)
            # Probably want mid-point rather than 0 for new dimensions.
            point = (
                point[:ndim]
                if ndim < len(point)
                else np.append(point, (0,) * (ndim - len(point)))
            )
            self.dims.set_point(range(ndim), point)
```

I'm not sure this is right, but I can try to think through it some more. In general this PR scares me a little bit just because `Dims` and `ViewerModel` are core components of napari, so changing them typically has lots of interesting side effects.

Reply:
Very confusing, I could have sworn I tried something similar and it did not work...
Great it does :D I made some adjustments to it to also cover the removal correctly.